### DISCUSSION TITLE
Promise.allSettled is not a function

### DISCUSSION THREAD

**Discussion by kmcconnell** [community/community#65175] [Copilot Conversations]
### Select Topic Area

Bug

### Body

We are frequently (to an unusable level) seeing the following error, both on latest release and pre-release in VS Code:

`[ERROR] [default] [2023-08-28T10:36:53.762Z] GitHub Copilot could not connect to server. Extension activation failed: "Promise.allSettled is not a function"`

VS Code: 
```
Version: 1.81.1 (user setup)
Commit: 6c3e3dba23e8fadc360aed75ce363ba185c49794
Date: 2023-08-09T22:22:42.175Z
Electron: 22.3.18
ElectronBuildId: 22689846
Chromium: 108.0.5359.215
Node.js: 16.17.1
V8: 10.8.168.25-electron.0
OS: Windows_NT x64 10.0.22621
```

GitHub Copilot: `v1.105.364`

I've signed out of Copilot in VS Code and signed back in. I've tried disabling and re-enabling, restarting VS Code, restarting the PC. Nothing solves this error. It is frequent enough that essentially we can't use Copilot. It is in a failed start more often than running.

**Comment by Akash1134** [comment]
Hi @kmcconnell ,

Thanks for reporting the issue and sorry for the inconvenience caused. 

I have asked my team internally to take a closer look at the error. We will get back to you soon.

**Comment by Akash1134** [comment]
Hi @kmcconnell , 

We believe that this error is from Copilot Chat extension. 

Could you confirm that the error is from Copilot extension or Copilot Chat in VSCode? 

Thanks!

**Reply by richgonyea** [reply]
@Akash1134 
I am seeing the same issue as the OP and I do have Copilot Chat extension installed.  I disabled Copilot Chat (and Copilot) extensions and the error is not reoccurring.


From VS Code help:
Version: 1.81.1 (user setup)
Date: 2023-08-09T22:22:42.175Z
Electron: 22.3.18
ElectronBuildId: 22689846
Chromium: 108.0.5359.215
Node.js: 16.17.1
V8: 10.8.168.25-electron.0
OS: Windows_NT x64 10.0.22621

**Reply by adilaittaleb** [reply]
@Akash1134 
 Indeed the problem is from the copilot chat extension, and I have noticed in my case that this error only occurs when I open a project containing XML files. I really hope this helps to determine the root cause of this error.

**Reply by totosan** [reply]
Same for me, still appears with that error, when opening XML files

**Comment by shadankarim** [comment]
Is there a resolution yet? I am still facing the issue

**Comment by fazaldegui** [comment]
I have encountered the same issue: GitHub Copilot Chat was working correctly until I opened an XML file in Visual Studio Code. From that moment on, the Copilot Chat panel started showing the following errors:

[error] TypeError: Promise.allSettled is not a function  
[error] TypeError: cf(...).then(...).catch(...).finally is not a function

After investigating the root cause, I found that the issue is due to an incompatibility between GitHub Copilot and the “XML Tools” extension installed in the environment. Once I uninstalled the “XML Tools” extension, Copilot started working correctly again, even when opening XML files.

**Comment by josecandel** [comment]
I had the same issue. Unable to log in to GitHub Copilot. I uninstalled the XML extension and everything went fine.

**Reply by dadidduekappa** [reply]
Can confirm!
